peachjam_search/views.py

```diff
@@ -318,6 +318,45 @@ def __init__(self, *args, **kwargs):
         self.index = get_search_indexes(self.document._index._name)
         self.search = self.search.index(self.index)
 
+    def get_translatable_fields(self, request):
+        # get language from request
+        language = request.LANGUAGE_CODE
+        # suffix for translatable fields e.g. court_en, court_fr etc
+        suffix = ""
+        if language in [lang for lang, _ in settings.LANGUAGES]:
+            suffix = "_" + language
+
+        translatable_facets = {
+            "court": {"field": f"court{suffix}", "options": {"size": 100}},
+            "nature": {"field": f"nature{suffix}", "options": {"size": 100}},
+            "registry": {"field": f"registry{suffix}", "options": {"size": 100}},
+            "order_outcome": {
+                "field": f"order_outcome{suffix}",
+                "options": {"size": 100},
+            },
+        }
+
+        translatable_filters = {
+            "court": f"court{suffix}",
+            "nature": f"nature{suffix}",
+            "registry": f"registry{suffix}",
+            "order_outcome": f"order_outcome{suffix}",
+        }
+
+        return translatable_facets, translatable_filters
+
+    def list(self, request, *args, **kwargs):
+        # get translatable facets and filters
+        translatable_facets, translatable_filters = self.get_translatable_fields(
+            request
+        )
+        # add translatable facets
+        self.faceted_search_fields.update(translatable_facets)
```

Comment:
This currently will update the field as it is on the original class. The next request will use the updated fields, which isn't correct. Instead, deep-copy `self.faceted_search_fields` and `self.filter_fields` and then update the copy.